File: problems/434-number-of-segments-in-a-string/solution.java
class Solution {
    public int countSegments(String s) {
        int count = 0;
        boolean inSegment = false;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ' ') {
                inSegment = false;
            } else if (!inSegment) {
                count++;
                inSegment = true;
            }
        }
        return count;
    }
}

File: problems/434-number-of-segments-in-a-string/solution.c
int countSegments(char* s) {
    int count = 0;
    int inSegment = 0;
    for (int i = 0; s[i] != '\0'; i++) {
        if (s[i] == ' ') {
            inSegment = 0;
        } else if (inSegment == 0) {
            count++;
            inSegment = 1;
        }
    }
    return count;
}